### PR TITLE
Enable pairing & contact recovery

### PR DESCRIPTION
**Pull request by cammellos**
This PR enables pairing outside of dev-mode and contact-recovery, which
is useful in the case a new device is added or re-installed.

To test pairing, the same applies to previous builds, but it everything should just be enabled by default.
To test contact recovery:

1) Create account on A1
2) Create account on B1
3) Create a group chat between A & B, exchange a few messages
4) Recover account on B2
5) Once the account is recovered, send a group chat message from A to B
6) You should see the message on B2

On nightly the last message should not be received by B2, you would have to create a group chat on B2 and include A, at that point the devices are able to communicate. 

It should help with https://github.com/status-im/status-react/issues/7362 although we still need logs to understand exactly the issue.

status: ready

### CODE REVIEW

**Review by jeluard — APPROVED** [review]

File: src/status_im/contact_recovery/core.cljs [hunk 1]
```diff
@@ -1,42 +1,66 @@
...
-  (when (show-contact-recovery-message? public-key)
-    (re-frame/dispatch [:contact-recovery.callback/show-contact-recovery-message public-key])))
+(fx/defn notify-user
+  "Send an empty message to the user, which will carry device information"
+  [cofx public-key]
+  (let [{:keys [web3]} (:db cofx)
+        current-public-key (accounts.db/current-public-key cofx)]
+    {:shh/send-direct-message
+     [{:web3 web3
+       :src current-public-key
```
Comment (dmitryn) [inline src/status_im/contact_recovery/core.cljs:56]: map key values align

**Review by dmitryn — APPROVED** [review]

File: src/status_im/pairing/core.cljs [hunk 2]
```diff
@@ -230,30 +226,26 @@
...
-            new-account  (merge-account (:account/account db) account)]
-        (fx/merge cofx
-                  {:db                 (assoc db
-                                              :contacts/contacts new-contacts
-                                              :account/account new-account)
-                   :data-store/base-tx [(data-store.accounts/save-account-tx new-account)]
-                   :data-store/tx      [(data-store.contacts/save-contacts-tx (vals new-contacts))]}
-                  #(when (:public? chat)
-                     (models.chat/start-public-chat % (:chat-id chat) {:dont-navigate? true})))))))
+  (when (and (= sender (accounts.db/current-public-key cofx)))
```
Comment (rasom) [inline src/status_im/pairing/core.cljs]: `and` is redundant

**Review by rasom — APPROVED** [review]
noop 9m

**Review by flexsurfer — APPROVED** [review]

**Comment by churik** [comment]
@cammellos feel free to ping me when it will be ready to test

**Comment by churik** [comment]
@cammellos tested scenario from Testing steps + general sync of contacts, user pic/name, 1-1, public and group chats between accounts Mac OSx / IOS / Android + sync with released version 0.9.32 (Aragon hotfix).

Everything looks fine to me, however I'm not sure how to deal with linux failed build.
Error:
```
17:27:08 The current environment doesn't contain the expected versions of node and/or yarn
17:27:08   - node:	expected	v10.12.0
17:27:08   		found		v10.15.0 (/usr/bin/node)
17:27:08   - yarn:	expected	1.13.0
17:27:09   		found		1.13.0 (/usr/bin/yarn)
17:27:09 Please open another console to reload the environment, and then run 'make setup' if necessary
```
Builds: https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7366/

What I've got from `#status-core-infra`:
see you are still using statusteam/status-build-linux:1.1.0  image , you might want to rebase

<blockquote><div><strong><a href="https://ci.status.im/job/status-react/job/prs/job/linux/job/PR-7366/">PR-7366 [status-react » prs » linux] [Jenkins]</a></strong></div></blockquote>

**Comment by cammellos** [comment]
@churik thanks, I have rebased now, let's see how it goes

**Comment by churik** [comment]
@cammellos what about link to FAQ that was discussed in test-team channel? don't see it here

**Comment by cammellos** [comment]
I am still waiting for the PR to be merged in the docs repo, so I will
probably add in a separate PR if it s ok.

**Comment by churik** [comment]
@cammellos OK, np

**Pull request merged**